cpp/arcticdb/pipeline/frame_utils.hpp

```diff
@@ -130,14 +131,19 @@ void aggregator_set_data(
                     ptr_data = flatten_tensor<PyObject*>(flattened_buffer, rows_to_write, tensor, slice_num, regular_slice_size);
 
                 auto none = py::none{};
+                // GIL will be acquired if there is a string that is not pure ASCII/UTF-8
+                // In this case a PyObject will be allocated by convert::py_unicode_to_buffer
+                // If such a string is encountered in a column, then the GIL will be held until that whole column has
+                // been processed, on the assumption that if a column has one such string it will probably have many.
+                std::unique_ptr<ScopedGILLock> scoped_gil_lock;
```

Comment:
I don't think we need this on the heap. Maybe std::optional would be a better choice.